### PR TITLE
Elasticsearch: decouple frontend dependencies

### PR DESCRIPTION
**Pull request by matyax**
Removes the following core dependencies:

- Flatten
- Datasource picker
- Log level extraction
- Variable regex

The last frontend dependency will be fixed by https://github.com/grafana/grafana/pull/80281

**Which issue(s) does this PR fix?**:

Part of https://github.com/grafana/grafana/issues/72632

**Special notes for your reviewer:**

Please check that:
- It doesn't break current functionality.

### CODE REVIEW

File: public/app/plugins/datasource/elasticsearch/utils.ts [hunk 1]
```diff
@@ -106,3 +108,64 @@ export const unsupportedVersionMessage =
...
 };
+
+/*
+ * This regex matches 3 types of variable reference with an optional format specifier
+ * There are 6 capture groups that replace will return
+ * \$(\w+)                                    $var1
+ * \[\[(\w+?)(?::(\w+))?\]\]                  [[var2]] or [[var2:fmt2]]
+ * \${(\w+)(?:\.([^:^\}]+))?(?::([^\}]+))?}   ${var3} or ${var3.fieldPath} or ${var3:fmt3} (or ${var3.fieldPath:fmt3} but that is not a separate capture group)
+ */
+export const variableRegex = /\$(\w+)|\[\[(\w+?)(?::(\w+))?\]\]|\${(\w+)(?:\.([^:^\}]+))?(?::([^\}]+))?}/g;
```
Comment (matyax) [inline public/app/plugins/datasource/elasticsearch/utils.ts:119]: Same simple approach: https://github.com/grafana/grafana/blob/main/public/app/plugins/datasource/prometheus/querybuilder/parsingUtils.ts#L31

File: public/app/plugins/datasource/elasticsearch/utils.ts [hunk 2]
```diff
@@ -106,3 +108,64 @@ export const unsupportedVersionMessage =
...
+export function dataFrameLogLevel(dataFrame: DataFrame): LogLevel {
+  const field = dataFrame.fields.find((f) => f.type === FieldType.number);
+  const level = field?.labels?.['level'] ?? '';
+  if (!level) {
+    return LogLevel.unknown;
+  }
+  return level in logLevelMap ? logLevelMap[level] : LogLevel.unknown;
+}
+
+export function flattenObject(target: Record<string, unknown> | unknown, prevKey = ''): Record<string, unknown> {
```
Comment (ivanahuckova) [inline public/app/plugins/datasource/elasticsearch/utils.ts:151]: Could we copy-paste here the logic from `flatten`. Cause this change is not equal, as it doesn't consider `maxDepth`. We are going to remove it with https://github.com/grafana/grafana/issues/65813 anyway, but as this PR shouldn't change the functionality, I would suggest to keep the original logic.

File: public/app/plugins/datasource/elasticsearch/datasource.ts [hunk 3]
```diff
@@ -31,7 +31,6 @@ import {
   SupplementaryQueryOptions,
   toUtc,
   AnnotationEvent,
-  FieldType,
```
Comment (ivanahuckova) [inline public/app/plugins/datasource/elasticsearch/datasource.ts:34]: Aren't all of these dependecies going to be removed with https://github.com/grafana/grafana/pull/80281? I think change in this file should be reverted as you have already PR that tackles it.

File: public/app/plugins/datasource/elasticsearch/utils.ts [hunk 4]
```diff
@@ -106,3 +108,64 @@ export const unsupportedVersionMessage =
...
+}
+
+export function dataFrameLogLevel(dataFrame: DataFrame): LogLevel {
+  const field = dataFrame.fields.find((f) => f.type === FieldType.number);
+  const level = field?.labels?.['level'] ?? '';
+  if (!level) {
+    return LogLevel.unknown;
+  }
+  return level in logLevelMap ? logLevelMap[level] : LogLevel.unknown;
+}
```
Comment (ivanahuckova) [inline public/app/plugins/datasource/elasticsearch/utils.ts:149]: We won't need any of these when https://github.com/grafana/grafana/pull/80281/

File: public/app/plugins/datasource/elasticsearch/utils.ts (hunk 2 above)
Comment (matyax) [reply public/app/plugins/datasource/elasticsearch/utils.ts:151]: Absolutely, will do that instead.

Comment (matyax) [reply public/app/plugins/datasource/elasticsearch/utils.ts:151]: Done.

**Comment by ivanahuckova** [comment]
I think that this one can be closed, right @matyax. 🙂

**Comment by matyax** [comment]
It certainly can!

**Pull request closed without merging**